src/Uno.UWP/System/Launcher.cs

```diff
@@ -1,105 +1,70 @@
-﻿#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
-
-using System;
+﻿using System;
 using System.Threading.Tasks;
-using Uno.Extensions;
-using Uno.Logging;
 using Windows.Foundation;
+using Uno.Logging;
+using Uno.Extensions;
 using Microsoft.Extensions.Logging;
-using System.Linq;
 using Windows.UI.Core;
 
-#if __ANDROID__
-using Android.Content;
-using Android.Content.PM;
-#endif
-#if __IOS__
-using UIKit;
-using AppleUrl = global::Foundation.NSUrl;
-#endif
-
 namespace Windows.System
 {
 	public static partial class Launcher
 	{
-		public static async Task<bool> LaunchUriAsync(Uri uri)
+		private const string MicrosoftUriPrefix = "ms-";
+
+		private static bool IsSpecialUri(Uri uri) => uri.Scheme.StartsWith(MicrosoftUriPrefix, StringComparison.InvariantCultureIgnoreCase);
+
+
+		public static IAsyncOperation<bool> LaunchUriAsync(Uri uri)
 		{
-			return await CoreDispatcher.Main.RunWithResultAsync(
-				CoreDispatcherPriority.Normal,
-				async () =>
-				{
-					try
-					{
-#if __IOS__
-						return UIKit.UIApplication.SharedApplication.OpenUrl(
-							new AppleUrl(uri.OriginalString));
-#elif __ANDROID__
-						var androidUri = Android.Net.Uri.Parse(uri.OriginalString);
-						var intent = new Intent(Intent.ActionView, androidUri);
+#if __IOS__ || __ANDROID__ || __WASM__
+
+			if (uri == null)
+			{
+				// this exception might not be in line with UWP which seems to throw AccessViolationException... for some reason.
+				throw new ArgumentNullException(nameof(uri));
+			}
 
-						if (Uno.UI.ContextHelper.Current == null)
-						{
-							throw new InvalidOperationException(
-								"LaunchUriAsync was called too early in application lifetime. " +
-								"App context needs to be initialized");
-						}
-						((Android.App.Activity)Uno.UI.ContextHelper.Current).StartActivity(intent);
+			if (!CoreDispatcher.Main.HasThreadAccess)
+			{
+				// LaunchUriAsync throws the following exception if used on UI thread on UWP
+				throw new InvalidOperationException("A method was called at an unexpected time.");
```

Comment:
I would suggest adding a error log here saying LaunchUriAsync should be call on the CoreDispatcher/MainThread.
It would help developers because the exception might not be obvious to everyone.

Reply:
Done 🙂